app/scripts/lib/session.js

```diff
@@ -41,7 +41,34 @@ define([
         // ignore the parse error.
       }
 
-      this.set(values);
+      // Update new values, without overwriting items on the prototype.
+      _.each(values, function (value, key) {
+        if (! Session.prototype.hasOwnProperty(key)) {
+          this[key] = value;
+        }
+      }, this);
+
+      return values;
+    },
+
+
+    /**
+     * Reload in-memory values based on what's currently in storage.
+     * @method reload
+     * This method can be used to pull in any changes to localStorage
```

Comment:
Also, can you make a note that this operation is destructive, as in, it'll remove any values currently in Session that are not also in sessionStorage or localStorage?